### DISCUSSION TITLE
Removing elements from IFC files in python

### DISCUSSION THREAD

**Discussion by elisemunche** [IfcOpenShell/IfcOpenShell#2875] [Q&A]
Hi! I am currently writing my master thesis and need some help with handling IFC files. I wish to manipulate a m IFC file so that it removes all the elements that are not load bearing (say for example every element besides IfcBeam and IfcColum). Then I need to read this IFC file in Grasshopper by using the plugin GeometryGym.

I have managed to write a code that removes everything except IfcBeam and IfcColumn in Python, but when I tried reading it in Grasshopper there are no values in the file. However, the new IFC file works in Solibri.

Here is my Python code:

_import ifcopenshell

f = ifcopenshell.open("My_file.ifc")
beams = f.by_type("IfcBeam")
columns=f.by_type("IfcColumn")

for instance in f:
if not instance.is_a("IfcBeam") and not instance.is_a("IfcColumn"):
f.remove(instance)
f.write("My_new_file.ifc")_

Does anyone have any experience with this and know what the problem with my code may be? 
Thank you so much in advance :D

**Comment by aothms** [comment]
Keep in mind that the instances in an IFC model are not just building elements, it's everything from properties to the individual points that make up the geometric representation of the elements. So what's left after this operation is more or less empty shells that will not import. Also all project and unit information will be gone.

A better start would be:

```
for instance in f.by_type('IfcBuildingElement'):
    ...
```

But even there it would be better to use higher level APIs such as https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/api/root/remove_product.py

**Comment by elisemunche** [comment]
Thank you so much! I will give it a new try 😃
